src/figuresClasses.ts

```diff
@@ -2,10 +1,83 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = `red` | `green` | `blue`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a < 1 || this.b < 1 || this.c < 1) {
+      throw new Error('Sides can`t be less than 1');
+    }
+
+    const p = this.a + this.b + this.c;
+
+    const maxValue = Math.max(this.a, this.b, this.c);
+
+    if (maxValue >= p - maxValue) {
+      throw new Error('It is not a triangle');
```

Comment:
[CODE STYLE] - The error message should be more descriptive. Consider: 'The provided side lengths do not form a valid triangle according to the triangle inequality theorem.'